src/bootstrap/src/ferrocene/run.rs

```diff
@@ -99,3 +100,139 @@ impl Step for TraceabilityMatrix {
         html_output
     }
 }
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash, PartialOrd, Ord)]
+pub(crate) struct GenerateCoverageReport;
+
+impl Step for GenerateCoverageReport {
+    type Output = ();
+    const ONLY_HOSTS: bool = true;
+    const DEFAULT: bool = true;
+
+    fn run(self, builder: &Builder<'_>) -> Self::Output {
+        builder.info(&format!("Generating coverage report"));
+
+        let coverage_report_data_dir =
+            env_llvm_profile_data_dir().unwrap_or_else(|| builder.out.join("coverage"));
+
+        let coverage_report_out_dir = builder.out.join("coverage_report");
+
+        let coverage_src_path = builder.out.join("../library/core").canonicalize().unwrap();
```

Comment:
`builder.src` points to the root of the repository, rather than going back with `..`.